src/main/java/com/example/alaa/university/repository/StudentRepository.java

```diff
@@ -1,4 +1,155 @@
+                    new java.sql.Timestamp(studentToBeAddedToDataBase.getBirthDate().toEpochMilli()));
+
+            preparedStatement.setTimestamp(9,
+                    new java.sql.Timestamp(studentToBeAddedToDataBase.getGraduatedDate().toEpochMilli()));
+
+            preparedStatement.setTimestamp(10,
+                    new java.sql.Timestamp(studentToBeAddedToDataBase.getRegistrationDate().toEpochMilli()));
+
+
+            return preparedStatement;
+        };
+
+        GeneratedKeyHolder generatedKeyHolder = new GeneratedKeyHolder();
+        template.update(preparedStatementCreator, generatedKeyHolder);
+        Number keyGenerated = generatedKeyHolder.getKey();
+        Long key = keyGenerated.longValue();
+        return get(key);
+    }
+
+
+    @Override
+    public Student update(Long id, Student updatedStudent) {
+        Student studentBeforeUpdated = this.get(id);
+        updatedStudent.getAddress().setId(studentBeforeUpdated.getAddress().getId());
+        Address newAddress = iAddressRepository.update(studentBeforeUpdated.getAddress().getId(),
+                updatedStudent.getAddress());
+
+
+        PreparedStatementCreator preparedStatementCreator = (Connection connection) -> {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement("UPDATE public.students\n" +
+                            "\tSET  name=?, gender=?, graduated=?, payment_fee=?, email=?, birth_date=?, graduated_date=?, registration_date=?\n" +
+                            "\tWHERE id=?");
+            preparedStatement.setString(1, updatedStudent.getName());
+            preparedStatement.setString(2, updatedStudent.getGender().toString());
+            preparedStatement.setBoolean(3, updatedStudent.getGraduated());
+            preparedStatement.setDouble(4, updatedStudent.getPaymentFee());
+            preparedStatement.setString(5, updatedStudent.getEmail());
+
+            preparedStatement.setTimestamp(6,
+                    new java.sql.Timestamp(updatedStudent.getBirthDate().toEpochMilli()));
+
+            preparedStatement.setTimestamp(7,
+                    new java.sql.Timestamp(updatedStudent.getGraduatedDate().toEpochMilli()));
+
+            preparedStatement.setTimestamp(8,
+                    new java.sql.Timestamp(updatedStudent.getRegistrationDate().toEpochMilli()));
+            preparedStatement.setLong(9, id);
+
+
+            return preparedStatement;
+        };
+        template.update(preparedStatementCreator);
+        return get(id);
+    }
+
+    @Override
+    public void delete(Long id) {
+        Long add=get(id).getAddress().getId();
+        PreparedStatementCreator preparedStatementCreator1=(Connection connection1)->{
+           PreparedStatement preparedStatement1=connection1.prepareStatement("UPDATE public.students\n" +
+                   "\tSET address_id=null, university_id=null\n" +
+                   "\tWHERE id=?" );
+
+           preparedStatement1.setLong(1,id);
```

Comment:
it is also not needed. Repository is straight forrward. It should take into account that students has a relation to adress or university